demos/defender/aws_iot_demo_defender.c

```diff
@@ -154,138 +92,216 @@ void _defenderCallback( void * param1,
+        {
+            IotLogError( "MQTT Initialization Failed." );
+            status = EXIT_FAILURE;
+        }
+    }
 
-    /* Set Defender callback. */
-    startInfo.callback = callback;
+    if( status == EXIT_SUCCESS )
+    {
+        /* Initialize Metrics. */
+        metricsInitStatus = IotMetrics_Init();
 
-    /* Invoke defender start API. */
-    defenderResult = AwsIotDefender_Start( &startInfo );
+        if( !metricsInitStatus )
+        {
+            IotLogError( "IOT Metrics Initialization Failed." );
+            status = EXIT_FAILURE;
+        }
+    }
 
-    return defenderResult;
-}
+    if( status == EXIT_SUCCESS )
+    {
+        /* Specify all metrics in "tcp connections" group */
+        defenderResult =
+            AwsIotDefender_SetMetrics( AWS_IOT_DEFENDER_METRICS_TCP_CONNECTIONS, AWS_IOT_DEFENDER_METRICS_ALL );
 
-/*-----------------------------------------------------------*/
+        if( defenderResult != AWS_IOT_DEFENDER_SUCCESS )
+        {
+            status = EXIT_FAILURE;
+        }
+    }
 
-#if _DEMO_WITH_SOCKET_CONNECTED_TO_ECHO_SERVER == 1
+    if( status == EXIT_SUCCESS )
+    {
+        /* Set metrics report period to 5 minutes(300 seconds) */
+        defenderResult = AwsIotDefender_SetPeriod( 300 );
+
+        if( defenderResult != AWS_IOT_DEFENDER_SUCCESS )
+        {
+            status = EXIT_FAILURE;
+        }
+    }
+
+    if( status == EXIT_SUCCESS )
+    {
+        /* Create MQTT Connection */
+        mqttStatus = _establishMqttConnection( pIdentifier,
+                                               pNetworkServerInfo,
+                                               pNetworkCredentialInfo,
+                                               pNetworkInterface,
+                                               &mqttConnection );
+
+        if( mqttStatus != IOT_MQTT_SUCCESS )
+        {
+            IotLogError( "Failed to Create MQTT Connection:%d", IotMqtt_strerror( mqttStatus ) );
+            IotMqtt_Cleanup();
+            defenderResult = AWS_IOT_DEFENDER_INTERNAL_FAILURE;
+        }
+        else
+        {
+            /* Initialize start info and call defender Start API */
+            startInfo.pClientIdentifier = pIdentifier;
+            startInfo.clientIdentifierLength = ( uint16_t ) strlen( pIdentifier );
+            startInfo.callback = callback;
+            startInfo.mqttConnection = mqttConnection;
+            defenderResult = AwsIotDefender_Start( &startInfo );
+        }
+
+        if( defenderResult == AWS_IOT_DEFENDER_SUCCESS )
+        {
+            /* Let it run for 3 seconds */
+            IotClock_SleepMs( 3000 );
+            /* Stop the defender agent. */
+            AwsIotDefender_Stop();
+            /* Disconnect MQTT */
+            IotMqtt_Disconnect( mqttConnection, false );
+        }
+    }
+
+    /* Cleanup. */
+    if( metricsInitStatus )
+    {
+        IotMetrics_Cleanup();
+    }
 
-    static Socket_t _createSocketToEchoServer()
```

Comment:
This was added to make demo report two TCP connections. Just FYI. I assume there was good team to remove it.

Reply:
It is removed so that demo remains platform independent.

Reply:
Make sense.